cmd/juju/action/common.go

```diff
@@ -146,15 +150,25 @@ func (c *runCommandBase) processOperationResults(ctx *cmd.Context, results *acti
 		}
 		return nil
 	}
-	return c.waitForTasks(ctx, tasks, info)
+	exitCode, err := c.waitForTasks(ctx, tasks, info)
+	if err != nil {
+		return errors.Trace(err)
+	} else if exitCode > 0 {
+		var plural string
+		if numTasks > 1 {
+			plural = "s"
+		}
+		return errors.Errorf("task%s failed with exit code: %d", plural, exitCode)
```

Comment:
Does a non-zero exit code mean that some tasks failed or that all tasks failed? If it's the former we should probably rephrase the error for the multiple task case to something like "some or all tasks failed with exit code: %d"

Reply:
`tasks` vs `task` should cover this.

Reply:
As written, the message does not make it clear if one task failed or whether all failed. So the user runs 4 tasks and one fails, they might assume that all failed with exit X.